tests/antares/services/api_services/test_study_api.py

```diff
@@ -330,3 +335,86 @@ def test_create_link_same_area(self):
         ):
             self.study.create_link(area_from=area, area_to=area)
+
+    def test_run_and_wait_antares_simulation(self):
+        parameters = AntaresSimulationParameters(solver=Solver.COIN, nb_cpu=2, unzip_output=True, presolve=False)
+
+        # patch simulates the repeating intervals so that we don't have to wait X seconds during the tests
+        with requests_mock.Mocker() as mocker, patch("time.sleep", return_value=None):
+            run_url = f"https://antares.com/api/v1/launcher/run/{self.study_id}"
+            job_id = "1234-g6z17"
+            mocker.post(run_url, json={"id": job_id}, status_code=200)
+
+            job_url = f"https://antares.com/api/v1/launcher/jobs/{job_id}"
+            output_id = "2024abcdefg-output"
+
+            # mock can take a list of responses that it maps one by one to each request, since we're doing multiple ones
+            response_list = [
+                {
+                    "json": {
+                        "id": job_id,
+                        "status": "pending",
+                        "launcher_params": dumps(parameters.to_api()),
+                    },
+                    "status_code": 200,
+                },
+                {
+                    "json": {
+                        "id": job_id,
+                        "status": "running",
+                        "launcher_params": dumps(parameters.to_api()),
+                    },
+                    "status_code": 200,
+                },
+                {
+                    "json": {
+                        "id": job_id,
+                        "status": "success",
+                        "launcher_params": dumps(parameters.to_api()),
+                        "output_id": output_id,
+                    },
+                    "status_code": 200,
+                },
+            ]
+            mocker.get(job_url, response_list)
+
+            tasks_url = "https://antares.com/api/v1/tasks"
+            task_id = "task-5678"
+            mocker.post(tasks_url, json=[{"id": task_id, "name": f"UNARCHIVE/{output_id}"}], status_code=200)
+
+            task_url = f"{tasks_url}/{task_id}"
+            mocker.get(task_url, json={"result": {"success": True}}, status_code=200)
+
+            job = self.study.run_antares_simulation(parameters)
+            assert isinstance(job, Job)
+            assert job.job_id == job_id
+            assert job.status == JobStatus.PENDING
+
+            self.study.wait_job_completion(job, time_out=10)
+
+            assert job.status == JobStatus.SUCCESS
+
+            response_list.pop()
+            mocker.get(job_url, response_list)
+            mocker.post(tasks_url, status_code=200)
+
+            job = self.study.run_antares_simulation(parameters)
+            with pytest.raises(SimulationTimeOutError):
+                self.study.wait_job_completion(job, time_out=2)
+
```

Comment:
Perhaps you can separate a bit more these steps with just a line `# =======` + a comment as it's a bit hard to read the whole test